app/services/id_token_builder.rb

```diff
@@ -57,14 +57,22 @@ def timestamp_claims
   def acr
     ial = identity.ial
     case ial
-    when Idp::Constants::IAL_MAX then Saml::Idp::Constants::IALMAX_AUTHN_CONTEXT_CLASSREF
+    when Idp::Constants::IAL_MAX then determine_ial_max_acr
```

Comment:
This is the only part I'm not sure about. I'm not sure whether we should return the requested value or the returned value 🤔

Reply:
@mitchellhenke i did it this way because it was explicitly called out in the ticket:

```
Expected: be prompted for password to decrypt bundle, returned to SP with verified PII with acr of IAL2
Actual: no password prompt, sent back with only email and acr of IAL0
```

i also double checked with @mmagsa who confirmed this was the desired workflow! (if you two decide otherwise, i'm happy to update.)

Reply:
ah yep, thanks!